core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -426,14 +435,13 @@ public Location lookup(Object keyBase, long keyOffset, int keyLength) {
    *
    * This is a thread-safe version of `lookup`, could be used by multiple threads.
    */
-  public void safeLookup(Object keyBase, long keyOffset, int keyLength, Location loc) {
+  public void safeLookup(Object keyBase, long keyOffset, int keyLength, Location loc, int hash) {
     assert(longArray != null);
 
     if (enablePerfMetrics) {
       numKeyLookups++;
     }
-    final int hashcode = HASHER.hashUnsafeWords(keyBase, keyOffset, keyLength);
-    int pos = hashcode & mask;
+    int pos = hash & mask;
```

Comment:
This doesn't work if hash is negative right? Not clear to me the new hash doesn't return negatives. assert this.